src/player.ts

```diff
@@ -229,7 +249,7 @@ export default class Player extends EventEmitter {
   // Not to confuse with the end of a round
   endTurn() {
     // Reset free terraforming steps
-    this.data.terraformSteps = 0;
+    this.data.terraformCostDiscount = 0;
```

Comment:
Actually this was my mistake, recently introduced. I thought it was `data.temporaryStep`. This line can be removed.

Reply:
ok.
the reset of `data.temporaryrange` and `data.temporarystep` are in the `player.build.219`

Reply:
actually, maybe they should be moved in `player.endTurn`. Because, currently, if they take the increase range / terraforming steps action and  can't build a mine, then the bonus would carry over for next turn. Which shouldn't happen.